node/src/blockchain/blockchain_interface.rs

```diff
@@ -718,14 +801,18 @@ mod tests {
             .into_iter()
             .map(|request| serde_json::from_slice(&request.body()).unwrap())
             .collect();
+        let bodies: Vec<String> = bodies
+            .into_iter()
+            .map(|b| serde_json::to_string(&b).unwrap())
+            .collect();
         assert_eq!(
-            format!("\"0x000000000000000000000000{}\"", &to[2..]),
-            bodies[0]["params"][0]["topics"][2].to_string(),
+            bodies,
```

Comment:
Is there any danger that this assert will be disrupted by JSON formatting? Say, if somehow we get spaces after the commas, or newlines, or indentation, or whatever? If so, it might be better to compare these two as parsed objects rather than JSON strings.

From Bert: you could eliminate a lot of the backslashes if you used `r#"` quotes.

Reply:
If this breaks because of json formatting I suspect that means the block chain service has changed and we'd want the broken test to warn us of that. The raw jsonrpc response is how we see the data and is very compact without extra spaces and new lines. Do you have another suggestion to do instead?

I saw Bert's comment regarding the `format!()` I could remove the format to eliminate the escaping but this means the `&to[2..]` address would have to be hardcoded in an additional place. Do you prefer this?

Reply:
I wouldn't give the life for being right on this but I think it would allow you to use `format!()` also with `r#"...text..."...text..."#` literals that contain double quotes and it even can have a placeholder in it like `{}` or `{:?}` and it should still work as long as you supply that many arguments as how many placeholders you put in there. You can try but maybe not worth it.